pkg/netceptor/netceptor.go

```diff
@@ -852,16 +852,28 @@ func (s *Netceptor) SubscribeRoutingUpdates() chan map[string]string {
 func (s *Netceptor) flood(message []byte, excludeConn string) {
 	s.connLock.RLock()
 	writeChans := make([]chan []byte, 0)
+	contexts := make([]context.Context, 0)
 	for conn, connInfo := range s.connections {
 		if conn != excludeConn {
 			writeChans = append(writeChans, connInfo.WriteChan)
+			contexts = append(contexts, connInfo.Context)
 		}
 	}
 	s.connLock.RUnlock()
+	wg := sync.WaitGroup{}
 	for i := range writeChans {
 		i := i
-		go func() { writeChans[i] <- message }()
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			select {
+			case writeChans[i] <- message:
+			case <-contexts[i].Done():
+				logger.Debug("connInfo cancelled during flood write")
+			}
+		}()
 	}
+	wg.Wait()
```

Comment:
this wait makes me a little nervous because it blocks the entire flood call until all of the writes are done, but at the same time not waiting on these go-routines also makes me nervous (i.e. go-routine leaks).

@ghjm any opinions on this? I could be convinced to just drop the wg.Wait() here

Reply:
I don't see the need for the WaitGroup at all.  We're not holding any locks and there's no cleanup code at the end.  I don't immediately recall any cases where a caller of flood() would depend on the messages being sent before the function returns, though it might be worthwhile to look at the places it's called and docker check this.

Also, it would be cleaner to just launch the goroutine inside the first loop, which would eliminate the need for the temporary data structures and the second loop.

Reply:
@ghjm okay I'm going to remove the waitgroup. I didn't have any evidence of this being a problem before, so no need to introduce a waitgroup here.